numpy/matrixlib/defmatrix.py

```diff
@@ -210,6 +215,9 @@ class matrix(N.ndarray):
     """
     __array_priority__ = 10.0
     def __new__(subtype, data, dtype=None, copy=True):
+        warnings.warn('the matrix subclass is not the recommended way to deal '
+                      'with linear algebra. Please adjust your code to use '
+                      'regular ndarray.', PendingDeprecationWarning)
```

Comment:
It would be a nice touch to add a URL pointing users to a longer description/discussion of why matrix is no longer recommended.

Reply:
This may also confuse users who aren't attempting to do linear algebra, but simply thought that `np.matrix` is the default ndarray constructor.

Reply:
Please also consider adding `stacklevel=1` to `warn()` such that the warning is raised from the calling function or `__main__` but not inside `np.matrix`.